src/halmos/sevm.py

```diff
@@ -2247,20 +2340,30 @@ def run(self, ex0: Exec) -> Tuple[List[Exec], Steps]:
                     ex.st.push(uint256(f_origin()))
                 elif opcode == EVM.ADDRESS:
                     ex.st.push(uint256(ex.this))
+                # TODO: define f_extcodesize for known addresses in advance
                 elif opcode == EVM.EXTCODESIZE:
-                    address = uint160(ex.st.pop())
-                    if address in ex.code:
-                        codesize = con(len(ex.code[address]))
+                    account = uint160(ex.st.pop())
+                    account_addr = self.resolve_address_alias(ex, account)
+                    if account_addr is not None:
+                        codesize = con(len(ex.code[account_addr]))
                     else:
-                        codesize = f_extcodesize(address)
+                        codesize = f_extcodesize(account)
                         if (
-                            address == hevm_cheat_code.address
-                            or address == halmos_cheat_code.address
+                            eq(account, hevm_cheat_code.address)
+                            or eq(account, halmos_cheat_code.address)
+                            or eq(account, console.address)
```

Comment:
do we need similar handling for address(0) here? Would it be helpful to add a constraint that it has no code? What about precompiles (I think addr <= 18)

Reply:
see todo comment. need more work around this. 

btw, why do we need special handling address(0), compared to other unknown addresses?

Reply:
will be done in a separate pr

Reply:
I'm not sure, intuitively I'm thinking that it may be useful to have special handling for address(0) as it is a "well known" address that for sure has no code